### PR TITLE
DS-3377 Solr query may be longer than allowed by the Web Container

### PR DESCRIPTION
**Pull request by nicolasschwab**
https://jira.duraspace.org/browse/DS-3377

Now only the UUID of the comm/coll the user has explicit ADMIN authorization
on are added to the "location" parameter

### CODE REVIEW

**Review by tdonohue — CHANGES_REQUESTED** [review]
Needs rebasing, as this has merge conflicts

**Comment by pnbecker** [comment]
Beside the necessary rebase, I also don't agree with the idea of this PR. In DSpace we use the concept of groups, subgroups and inheritance of permissions from groups to subgroups to EPersons.

File: dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java [hunk 1]
```diff
@@ -711,52 +711,53 @@ public String createLocationQueryForAdministrableItems(Context context)
...
-                    }
-                }
-                locationQuery.append(")");
+            if(communitiesPolicies.size() != 0 || collectionsPolicies.size() != 0){
+            	
+            	locationQuery.append("location:( ");
+            	
+            	locationQuery.append(this.getLocationOfEachResourcePolicy(communitiesPolicies, context));
+            	
+            	locationQuery.append(this.getLocationOfEachResourcePolicy(collectionsPolicies, context));
```
Comment (terrywbrady) [inline dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java]: If both communitiesPolicies and collectionsPolicies exist, will the " OR " be inserted between the 2 lists?

**Review by terrywbrady — CHANGES_REQUESTED** [review]
See question about inserting OR between a list of communities and a list of collections.

File: dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java (hunk 1 above)
Comment (terrywbrady) [inline dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java]: I validated the issue that I reported above.  When a user has ADMIN rights to both collections and communities, the location search is corrupted.

> )+OR+location:(+mfc564c62-b178-4d8b-8789-995e3e7e5377+OR+m40249564-ea3a-4bdc-86fd-21cf6b554187ld65b663c-647c-4a4f-9737-671af7f0c8ab+)

Comment (terrywbrady) [inline dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java]: Aside from the issue reported above, when I test with this patch, the same number of objects are returned when only the root location is included in the search, so I think this PR should be merged once the bug is fixed.

Comment (terrywbrady) [inline dspace-api/src/main/java/org/dspace/discovery/SolrServiceImpl.java]: The following commit should resolve the bug: https://github.com/Georgetown-University-Libraries/DSpace/commit/4454550911781a293fdd383d37b9b81602955c3e

**Comment by tdonohue** [comment]
Note: there is an alternative PR which simply changes all Solr requests to HTTP POST, see #2045   Unless anyone feels differently, we may be able to close this PR in favor of that new one.

**Comment by tdonohue** [comment]
Closing, as this was fixed in #2045

**Pull request closed without merging**